src/deadline/keyshot_adaptor/KeyShotClient/keyshot_handler.py

```
@@ -125,3 +150,54 @@ def set_render_options(self, data: dict) -> None:
         """
         if "render_options" in data:
             self.render_kwargs["render_options"] = data["render_options"]
+
+    def set_override_render_device(self, data: dict) -> None:
+        """
+        Sets the override render device flag
+
+        Args:
+            data (dict): The data given from the Adaptor. Keys expected: ['override_render_device']
+        """
+        self.render_kwargs["override_render_device"] = data["override_render_device"]
+
+    def set_render_device(self, data: dict) -> None:
+        """
+        Sets the render engine (CPU or GPU) based on the parameter value.
+
+        Args:
+            data (dict): The data given from the Adaptor. Keys expected: ['render_device']
+
+        Raises:
+            RuntimeError: If GPU rendering is requested but no compatible GPU is available
+        """
+        override_render_device = self.render_kwargs["override_render_device"]
+
+        current_engine = lux.getRenderEngine()
+        is_gpu = current_engine in [lux.RENDER_ENGINE_PRODUCT_GPU, lux.RENDER_ENGINE_INTERIOR_GPU]
+        self.render_kwargs["current_device"] = "GPU" if is_gpu else "CPU"
+
+        if not override_render_device:
+            self.render_kwargs["render_device"] = "GPU" if is_gpu else "CPU"
+        else:
+            self.render_kwargs["render_device"] = data["render_device"]
+
+        if self.render_kwargs["render_device"] == "GPU":
+            try:
+                if not is_gpu:
+                    engine_map = {
+                        lux.RENDER_ENGINE_PRODUCT: lux.RENDER_ENGINE_PRODUCT_GPU,
+                        lux.RENDER_ENGINE_INTERIOR: lux.RENDER_ENGINE_INTERIOR_GPU,
+                    }
+                    if current_engine in engine_map:
+                        lux.setRenderEngine(engine_map[current_engine])
+            except Exception:
```

nit: we only expect this exception to occur during the `lux.setRenderEngine()` call, right? Can we minimize the scope of the try-except to wrap around only that line instead to be more precise and avoid swallowing unexpected errors?

---

Done!